unofficial/c511027106.lua

```diff
@@ -0,0 +1,102 @@
+--ＺＷ－風神雲龍剣
+--ZW - Tornado Bringer (anime)
+--Script by Rundas
+local s,id=GetID()
+function s.initial_effect(c)
+	--equip
+	local e1=Effect.CreateEffect(c)
+	e1:SetDescription(aux.Stringid(id,0))
+	e1:SetType(EFFECT_TYPE_IGNITION)
+	e1:SetProperty(EFFECT_FLAG_CARD_TARGET)
+	e1:SetCategory(CATEGORY_EQUIP)
+	e1:SetRange(LOCATION_HAND+LOCATION_MZONE)
+	e1:SetTarget(s.eqtg)
+	e1:SetOperation(s.eqop)
+	c:RegisterEffect(e1)
+	--protecc
+	local e2=Effect.CreateEffect(c)
+	e2:SetDescription(aux.Stringid(id,1))
+	e2:SetType(EFFECT_TYPE_EQUIP)
+	e2:SetCode(EFFECT_CANNOT_BE_EFFECT_TARGET)
+	e2:SetProperty(EFFECT_FLAG_IGNORE_IMMUNE)
+	e2:SetValue(s.val)
+	c:RegisterEffect(e2)
+    --destroy replace
+    local e3=Effect.CreateEffect(c)
+    e3:SetType(EFFECT_TYPE_FIELD+EFFECT_TYPE_CONTINUOUS)
+    e3:SetCode(EFFECT_DESTROY_REPLACE)
+    e3:SetRange(LOCATION_SZONE)
+    e3:SetTarget(s.reptg)
+    e3:SetValue(s.repval)
+    e3:SetOperation(s.repop)
+    c:RegisterEffect(e3)
+end
+
+s.listed_names={56840427}
+
+--equip
+
+function s.filter(c)
+	return c:IsFaceup() and c:IsCode(56840427)
+end
+
+function s.eqtg(e,tp,eg,ep,ev,re,r,rp,chk,chkc)
+	if chkc then return chkc:IsLocation(LOCATION_MZONE) and chkc:IsControler(tp) and s.filter(chkc) end
+	if chk==0 then return Duel.GetLocationCount(tp,LOCATION_SZONE)>0
+		and Duel.IsExistingTarget(s.filter,tp,LOCATION_MZONE,0,1,nil) end

```

Comment:
This is not a necessary change, but "s.filter" could be removed completely with the usage of "aux.FilterFaceupFunction".

Reply:
You forgot to use "aux.FilterFaceupFunction" in the "Duel.IsExistingTarget" line. Also, the face-up and code checks are still required in the "if chkc.." line.